[Hunter] Checklist updating and general code update

Pewtro opened the pull request:
Minor clean up in the code and adding a bit more into the checklist

anom0ly commented on src/Parser/Hunter/Marksmanship/Modules/Features/VulnerableUptime.js:
```diff
@@ -14,25 +14,35 @@ class VulnerableUpTime extends Analyzer {
-          .regular(recommended - 0.05).major(recommended - 0.10);
-      });
+    when(this.uptimeThreshold).addSuggestion((suggest, actual, recommended) => {
+      return suggest(<span>Your <SpellLink id={SPELLS.VULNERABLE.id} /> uptime can be improved. Make sure you use Windburst or Marked Shot to open the Vulnerable Window to maximize damage with your Aimed Shots</span>)
```
Since you are going back to rework these suggestions, I would update any `<span>` tags with `<Wrapper>` just to be consistent with the new style.

anom0ly commented on src/Parser/Hunter/Marksmanship/Modules/Spells/Trueshot.js:
```diff
@@ -188,45 +184,30 @@ class Trueshot extends Analyzer {
-          .major(recommended - 0.5);
-      });
+    when(this.aimedShotThreshold).addSuggestion((suggest, actual, recommended) => {
+      return suggest(<span>You only cast {actual} <SpellLink id={SPELLS.AIMED_SHOT.id} />s inside your average <SpellLink id={SPELLS.TRUESHOT.id} /> window. This is your only DPS cooldown, and it's important to maximize it to it's fullest potential by getting as many Aimed Shot squeezed in as possible, while still making sure that they are all within <SpellLink id={SPELLS.VULNERABLE.id} />. <br / ...
```
Same as above, use `<Wrapper>` instead of `<span>`

anom0ly commented on src/Parser/Hunter/Marksmanship/Modules/Spells/Trueshot.js:
```diff
@@ -188,45 +184,30 @@ class Trueshot extends Analyzer {
+        .recommended(`>${recommended} is recommended`);
+    });
+    when(this.focusThreshold).addSuggestion((suggest, actual, recommended) => {
+      return suggest(<span>You started your average <SpellLink id={SPELLS.TRUESHOT.id} /> at {actual} focus, try and pool a bit more before casting <SpellLink id={SPELLS.TRUESHOT.id} />. This can be done through casting an additional <SpellLink id={SPELLS.ARCANE_SHOT.id} /> or by monitoring the cooldown of <SpellLink id={SPELLS.TRUESHOT.id} /> and adjusting play to ensure your focus won't be depl ...
```
Same as above, use `<Wrapper>` instead of `<span>`

anom0ly commented on src/Parser/Hunter/Marksmanship/Modules/Spells/Trueshot.js:
```diff
@@ -188,45 +184,30 @@ class Trueshot extends Analyzer {
+        .recommended(`>${recommended} is recommended`);
+    });
+    when(this.executeTrueshotThreshold).addSuggestion((suggest, actual) => {
+      return suggest(<span>You should make sure to have atleast 1 <SpellLink id={SPELLS.TRUESHOT.id} /> cast during execute (where you are buffed by <SpellLink id={SPELLS.BULLSEYE_TRAIT.id} />) to get as much out of <SpellLink id={SPELLS.TRUESHOT.id} /> as possible.</span>)
```
Same as above, use `<Wrapper>` instead of `<span>`

anom0ly commented on src/Parser/Hunter/Marksmanship/Modules/Spells/Trueshot.js:
```diff
@@ -188,45 +184,30 @@ class Trueshot extends Analyzer {
+        .recommended(`casting atleast 1 trueshot in execute is recommended`);
+    });
+    when(this.uptimeThreshold).addSuggestion((suggest, actual) => {
+      return suggest(<span>You should make sure to utilise every possible second of <SpellLink id={SPELLS.TRUESHOT.id} /> uptime as you can. Remember to cast it atleast 15 seconds before the boss dies, so you don't lose out on valuable time, aswell as remember to not cast it until the boss has been engaged.</span>)
```
Same as above, use `<Wrapper>` instead of `<span>`

anom0ly commented on src/Parser/Hunter/Marksmanship/Modules/Talents/AMurderOfCrows.js:
```diff
@@ -128,14 +128,12 @@ class AMurderOfCrows extends Analyzer {
-          .regular(recommended);
-      });
+    when(this.shouldHaveSavedThreshold).addSuggestion((suggest) => {
+      return suggest(<span>You should <b>generally</b> save <SpellLink id={SPELLS.A_MURDER_OF_CROWS_TALENT_SHARED.id} /> when the boss has under 25% hp so that it is ready to use when the boss hits 20% and you can start getting <SpellLink id={SPELLS.BULLSEYE_BUFF.id} /> quicker.</span>)
```
Same as above, use `<Wrapper>` instead of `<span>`

anom0ly commented on src/Parser/Hunter/Marksmanship/Modules/Talents/PatientSniper/PatientSniperDetails.js:
```diff
@@ -64,22 +64,12 @@ class PatientSniperDetails extends Analyzer {
-          .major(major);
-      });
+    when(this.patientSniperDamageThresholds).addSuggestion((suggest, actual, recommended) => {
+      return suggest(<span><SpellLink id={SPELLS.PATIENT_SNIPER_TALENT.id} /> increases the damage of your <SpellLink id={SPELLS.AIMED_SHOT.id} /> or <SpellLink id={SPELLS.PIERCING_SHOT_TALENT.id} /> the later you fire them inside <SpellLink id={SPELLS.VULNERABLE.id} />. While this isn't worth waiting for, it looks like you're shooting your Aimed Shots / Piercing Shots too soon, try and use <Spel ...
```
Same as above, use `<Wrapper>` instead of `<span>`

Pewtro commented:
Went through Marksmanship, BeastMastery and Shared folders and found all uses of span and changed all the ones that needed changing. I'm a bit in doubt of the ones in TitansThunder, AMurderOfCrows, DireBeast but they work still atleast. 

Also removed Titan's Thunder from GCD (this is also fixed in abilities refactor) - and removed Golganneths cast from BM cooldown window.

Pewtro commented:
Should probably hold off with merging this until #1086 is in, I'll update this PR as needed to accommodate any changes that may occur.

Pewtro commented:
This is now updated with the abilities refactor in mind

anom0ly commented:
Thanks!
